components/CreateOrUpdateRecipeForm.js

```diff
@@ -0,0 +1,212 @@
+import styled from "styled-components";
+import { useRouter } from "next/router";
+import Link from "next/link";
+
+export default function CreateOrUpdateRecipeForm({
+  errorMessage,
+  inputValidation,
+  recipeDetails,
+  onUpdateRecipe,
+  onHandleSubmit,
+  isUpdate,
+}) {
+  const router = useRouter();
+
+  return (
+    <>
+      <h1>{isUpdate ? "Update Recipe" : "Create new recipe"}</h1>
+
+      {errorMessage ? (
+        <StyledErrorMessageContainer>
+          <StyledErrorMessage>{errorMessage}</StyledErrorMessage>
+        </StyledErrorMessageContainer>
+      ) : (
+        ""
+      )}
+      <StyledForm onSubmit={isUpdate ? onUpdateRecipe : onHandleSubmit}>
+        <label htmlFor="title">Title:</label>
+        <StyledTitleInput
+          type="text"
+          id="title"
+          name="title"
+          required
+          maxLength={75}
+          autoFocus
+          $titleInput={inputValidation}
+          defaultValue={isUpdate ? recipeDetails.name ?? "" : null}
```

Comment:
```suggestion
          defaultValue={recipeDetails.name}
```

Like I suggested above: With the prop `recipeDetails = {}` you can set an empty object as default value when no recipeDetails prop is given and like this you won't need to check if isUpdate is true or not.

Same for all your other input's defaultValues of course :)